### DISCUSSION TITLE
Emit help

### DISCUSSION THREAD

**Discussion by itsfuad** [socketio/socket.io#4924] [General] [answered]
I've gone through **emit cheat sheet** v3 and v4 but some things are not working properly. May be I misunderstood something. 

For broadcasting an event to all clients in a room except sender, I used 
```js
socket.broadcast.emit('server_message', { text: `${name} joined the that🔥`, id: crypto.randomUUID() }, 'join');
```

But the message event is captured by every clients in every room and without room.

We have
```js
io.to('room-name');
```
and
```js
socket.to('room-name');
```

I need a clarification on rooms, all-clients, only-sender, except-sender, all-clients-in-room-[..], etc.

I'm using **Deno** as the server runtime

**Comment by darrachequesne** [comment, marked as answer]
Hi!

| Method                      | Description                                                           |
|-----------------------------|-----------------------------------------------------------------------|
| `io.emit()`                 | to all connected clients                                              |
| `socket.broadcast.emit()`   | to all connected clients, except the sender                           |
| `io.to("room1").emit()`     | to all connected clients in the room named "room1"                    |
| `socket.to("room1").emit()` | to all connected clients in the room named "room1", except the sender |

Basically, rooms are filters that you apply when broadcasting an event.
